extensions/amp-base-carousel/1.0/dimensions.js

```diff
@@ -0,0 +1,228 @@
+ */
+export function getDimension(axis, el) {
+  const {
+    top,
+    bottom,
+    height,
+    left,
+    right,
+    width,
+  } = el./*OK*/ getBoundingClientRect();
+
+  return {
+    start: axis == Axis.X ? left : top,
+    end: axis == Axis.X ? right : bottom,
+    length: axis == Axis.X ? width : height,
+  };
+}
+
+/**
+ * @param {!Axis} axis The axis to get the center point for.
+ * @param {!Element} el The Element to get the center point for.
+ * @return {number} The center point.
+ */
+export function getCenter(axis, el) {
+  const {start, end} = getDimension(axis, el);
+  return (start + end) / 2;
+}
+
+/**
+ * @param {!Axis} axis The axis to get the start point for.
+ * @param {!Element} el The Element to get the start point for.
+ * @return {number} The start point.
+ */
+export function getStart(axis, el) {
+  const {start} = getDimension(axis, el);
+  return start;
+}
+
+/**
+ * @param {!Axis} axis The Axis to get the position for.
+ * @param {!Alignment} alignment The Alignment to get the position for.
+ * @param {!Element} el The Element to get the position for.
+ * @return {number} The position for the given Element along the given axis for
+ *    the given alignment.
+ */
+export function getPosition(axis, alignment, el) {
+  return alignment == Alignment.START
+    ? getStart(axis, el)
+    : getCenter(axis, el);
+}
+
+/**
+ * @param {!Axis} axis The axis to check for overlap.
+ * @param {!Element} el The Element to check for overlap.
+ * @param {number} position A position to check.
+ * @return {boolean} If the element overlaps the position along the given axis.
+ */
+export function overlaps(axis, el, position) {
+  const {start, end} = getDimension(axis, el);
+  // Ignore the end point, since that is shared with the adjacent Element.
+  return start <= position && position < end;
+}
+
+/**
+ * @param {!Axis} axis The axis to align on.
+ * @param {!Alignment} alignment The desired alignment.
+ * @param {!Element} container The container to align against.
+ * @param {!Element} el The Element get the offset for.
+ * @return {number} How far el is from alignment, as a percentage of its length.
+ */
+export function getPercentageOffsetFromAlignment(
+  axis,
+  alignment,
+  container,
+  el
+) {
+  const elPos = getPosition(axis, alignment, el);
+  const containerPos = getPosition(axis, alignment, container);
+  const {length: elLength} = getDimension(axis, el);
+  return (elPos - containerPos) / elLength;
+}
+
+/**
+ * Finds the index of a child that overlaps a point within the parent,
+ * determined by an axis and alignment. A startIndex is used to look at the
+ * children that are more likely to overlap first.
+ * @param {!Axis} axis The axis to look along.
+ * @param {!Alignment} alignment The alignment to look for within the parent
+ *    container.
+ * @param {!Element} container  The parent container to look in.
+ * @param {!Array<!Element>} children The children to look among.
```

Comment:
`container.children` is not an `Array`. I think it's a `HTMLCollection` or make be `IArrayLike` or something.

Reply:
[Confirmed](https://developer.mozilla.org/en-US/docs/Web/API/ParentNode/children) it is an `HTMLCollection` and updated accordingly